### PR TITLE
Disable Cacheops by default.

### PR DESCRIPTION
**Pull request by jathanism**
# Closes: #1721 
# What's Changed

- `CACHEOPS_ENABLED` now defaults to `False`
- Documentation updated to reflect the new default value of `CACHEOPS_ENABLED`
- Also remove instances of Cacheops `@file_cache.cache` as they are no longer necessary as of the `Job` model rewrite and `JobResult` refactoring from v1.4.0.
- Added `v2 TODO` comments to remind us to reap `cacheops` in v2 release.

# TODO

- [ ] Explanation of Change(s)
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Comment by HanlinMiao** [comment]
Is this [issue](https://github.com/nautobot/nautobot/issues/2324) resolved by this PR? Should we test it?

File: nautobot/core/templates/nautobot_config.py.j2 [hunk 1]
```diff
@@ -113,7 +113,7 @@ ALLOWED_URL_SCHEMES = (
 CACHEOPS_DEFAULTS = {"timeout": int(os.getenv("NAUTOBOT_CACHEOPS_TIMEOUT", "900"))}
 
 # Set to False to disable caching with cacheops. (Default: True)
-CACHEOPS_ENABLED = is_truthy(os.getenv("NAUTOBOT_CACHEOPS_ENABLED", "True"))
+CACHEOPS_ENABLED = is_truthy(os.getenv("NAUTOBOT_CACHEOPS_ENABLED", "False"))
```
Comment (glennmatthews) [inline nautobot/core/templates/nautobot_config.py.j2:116]: So users with existing `nautobot_config.py` will still be defaulting to `True` unless they regenerate their configs. Can we call that out in the release overview that this is something they may need to manually change when upgrading?

File: nautobot/docs/additional-features/caching.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ Nautobot makes extensive use of caching; this is not a simple topic but it's a u
...
 
-Nautobot supports database query caching using [`django-cacheops`](https://github.com/Suor/django-cacheops) and Redis. When a query is made, the results are cached in Redis for a short period of time, as defined by the [`CACHEOPS_DEFAULTS`](../configuration/optional-settings.md#cacheops_defaults) parameter (15 minutes by default). Within that time, all recurrences of that specific query will return the pre-fetched results from the cache. Caching can be completely disabled by toggling [`CACHEOPS_ENABLED`](../configuration/optional-settings.md#cacheops_enabled) to `False` (it is `True` by default).
+Nautobot supports database query caching using [`django-cacheops`](https://github.com/Suor/django-cacheops) and Redis. When a query is made, the results are cached in Redis for a short period of time, as defined by the [`CACHEOPS_DEFAULTS`](../configuration/optional-settings.md#cacheops_defaults) parameter (15 minutes by default). Within that time, all recurrences of that specific query will return the pre-fetched results from the cache. Caching can be enabled by toggling [`CACHEOPS_ENABLED`](../configuration/optional-settings.md#cacheops_enabled) to `True` (it is `False` by default).
```
Comment (glennmatthews) [inline nautobot/docs/additional-features/caching.md:13]: Let's rephrase this paragraph a bit more thoroughly - perhaps:

```suggestion
Nautobot optionally supports database query caching using [`django-cacheops`](https://github.com/Suor/django-cacheops) and Redis. Caching can be enabled by toggling [`CACHEOPS_ENABLED`](../configuration/optional-settings.md#cacheops_enabled) to `True` (it is `False` by default). When caching is enabled, and a query is made, the results are cached in Redis for a short period of time, as defined by the [`CACHEOPS_DEFAULTS`](../configuration/optional-settings.md#cacheops_defaults) parameter (15 minutes by default). Within that time, all recurrences of that specific query will return the pre-fetched results from the cache.

!!! caution "Changed in version 1.5.0"
    Query caching is now disabled by default, and will be removed as a supported option in a future release.
```

File: nautobot/core/templates/nautobot_config.py.j2 (hunk 1 above)
Comment (jathanism) [reply nautobot/core/templates/nautobot_config.py.j2:116]: Yes.

**Comment by jathanism** [comment]
> Is this [issue](https://github.com/nautobot/nautobot/issues/2324) resolved by this PR? Should we test it?

Likely any cacheops-related issues would be addressed by this, yes!

File: nautobot/extras/jobs.py [hunk 3]
```diff
@@ -940,7 +939,8 @@ def get_jobs():
     return jobs
 
 
-@file_cache.cached(timeout=60)
+# @file_cache.cached(timeout=60)
+# @cache.memoize(timeout=60)
```
Comment (bryanculver) [inline nautobot/extras/jobs.py:943]:
```suggestion
```

File: nautobot/extras/utils.py [hunk 4]
```diff
@@ -29,7 +28,6 @@
 logger = logging.getLogger(__name__)
 
 
-@file_cache.cached(timeout=60)
```
Comment (bryanculver) [inline nautobot/extras/utils.py:32]: Since we aren't removing any other prefetches, etc. is there a special reason to stop these cache entries now?

**Review by bryanculver — APPROVED** [review]

**Review by glennmatthews — APPROVED** [review]

File: nautobot/extras/jobs.py (hunk 3 above)
Comment (jathanism) [reply nautobot/extras/jobs.py:943]: Dang I thought I got rid of that.

File: nautobot/extras/utils.py (hunk 4 above)
Comment (jathanism) [reply nautobot/extras/utils.py:32]: In my testing to compare solutions, I was not able to reproduce the original issue with cascading calls in the job results list view. As far as I can tell, our refactoring of `Job` internals with the concrete model work made this problem go away and therefore the file caching is no longer necessary.

**Pull request merged**